lib/promise.mli

```diff
@@ -0,0 +1,13 @@
+type 'a promise
+val _Promise : 'a
+val _catch : 'a promise Js.t -> ('b -> 'a promise Js.t) -> 'a promise Js.t
+val _then :
+  'a promise Js.t ->
+  ?catch:('c -> 'b promise Js.t) ->
+  ('a -> 'b promise Js.t) -> 'b promise Js.t
+val resolve_value : 'a -> 'a promise Js.t
+val resolve_promise : 'a promise Js.t -> 'a promise Js.t
+val reject_value : 'a -> 'a promise Js.t
```

Comment:
I think this should be `'a -> 'b promise Js.t`, in the style of this binding. The rejection type is not related to the fulfillment type. The same of `reject_promise`.